src/server/project.ts

```diff
@@ -90,6 +90,18 @@ namespace ts.server {
         }
     }
 
+    export interface PluginCreateInfo {

```

Comment:
+1, plugins should be monitoring cancellation requests similar to normal LS methods